core/templates/cowdata.h

```diff
@@ -220,26 +220,63 @@ class CowData {
 	Error resize(Size p_size);
 
 	_FORCE_INLINE_ void remove_at(Size p_index) {
-		ERR_FAIL_INDEX(p_index, size());
-		T *p = ptrw();
 		Size len = size();
-		for (Size i = p_index; i < len - 1; i++) {
-			p[i] = p[i + 1];
+		ERR_FAIL_INDEX(p_index, len);
+
+		if (len == 1) {
+			// Removing the only element.
+			clear();
+			return;
 		}
 
-		resize(len - 1);
+		T *ptr = ptrw();
+		T *ptr_idx = ptr + p_index;
+		if constexpr (std::is_trivially_destructible_v<T>) {
```

Comment:
I think this check needs to be trivially destructible *and* constructable since we are moving.

Reply:
I don't think so. No construction is taking place, so trivial constrictability is unnecessary. In the case of trivial destructability, but nontrivial constructability:
- all objects are relocated
- the array is shrunk
    - the final object is trivially destructible, so the memory is merely reinterpreted without any destructor calls

Reply:
That problem would be better covered under nontrivial relocatability, as discussed below, so I'm closing this one.